tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -662,84 +862,337 @@ TEST(RDFVary, VaryDisplay) // TEST instead of TEST_P because Display is single-t
    // must update this test when https://github.com/root-project/root/issues/9894 is addressed
 }
 
-TEST_P(RDFVary, DISABLED_VaryFill)
-{
+struct Jet {
+   double a, b;
+};
 
-}
+struct CustomFiller {
 
-TEST_P(RDFVary, DISABLED_VaryGraph)
-{
+   TH2D h{"", "", 10, 0, 10, 10, 0, 10};
 
-}
+   void Fill(const Jet &j) { h.Fill(j.a, j.b); }
 
-TEST_P(RDFVary, DISABLED_VaryGraphAsymmErrors)
-{
+   void Merge(const std::vector<CustomFiller *> &jets)
+   {
+      TList l;
+      for (auto *j : jets)
+         l.Add(&j->h);
+      h.Merge(&l);
+   }
+};
 
-}
+TEST_P(RDFVary, VaryCustomObject)
+{
+   // this object can be filled
+   auto h = ROOT::RDataFrame(10)
+                   .Define("Jet",
+                           [] {
+                              return Jet{1., 2.};
+                           })
+                   .Vary(
+                      "Jet",
+                      [] {
+                         return ROOT::RVec<ROOT::RVecD>{{}, {4., 5.}};
+                      },
+                      {}, 2)
+                   .Fill<Jet>(CustomFiller{}, {"Jet"});
 
-TEST_P(RDFVary, DISABLED_VaryHisto2D)
-{
+   auto hs = VariationsFor(h);
 
-}
+   EXPECT_DOUBLE_EQ(h->GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(h->GetMeanX(), 1.);
+   EXPECT_DOUBLE_EQ(h->GetMeanY(), 2.);
 
-TEST_P(RDFVary, DISABLED_VaryHisto3D)
-{
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetMeanX(), 1.);
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetMeanY(), 2.);
+
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetMeanX(), 0.);
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetMeanY(), 0.);
 
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetMeanX(), 4.);
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetMeanY(), 5.);
 }
 
-TEST_P(RDFVary, DISABLED_VaryHistoND)
+TEST_P(RDFVary, VaryFill)
 {
-
+   SimpleFiller sf;
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [] { return 42.; })
+               .Vary(
+                  "x",
+                  [](double x) {
+                     return ROOT::RVecD{x - 1., x + 1.};
+                  },
+                  {"x"}, 2)
+               .Fill<double>(sf, {"x"});
+   EXPECT_THROW(VariationsFor(h), std::logic_error); // FIXME not (yet) implemented
```

Comment:
please write the tests for what we expect to happen :smile: If I'm reading the code correctly we should get histograms filled with 42 (nominal), 41 (x:0) and 43 (x:1)